engine/gamesys/src/gamesys/resources/res_texture.h

```diff
@@ -20,6 +20,22 @@
 
 namespace dmGameSystem
 {
+    struct ResTextureUploadParams
+    {
+        uint16_t m_X;
+        uint16_t m_Y;
+        uint8_t  m_MipMap               : 5;
+        uint8_t  m_UploadSpecificMipmap : 1;
+        uint8_t  m_SubUpdate            : 1;
+        uint8_t                         : 1;
+    };
+
+    struct ResTextureReCreateParams
+    {
+        void*                  m_TextureImage;
+        ResTextureUploadParams m_UploadParams;
+    };
+
```

Comment:
Instead of having both data and parameters on how to upload data in the same struct, this is now split into two structs so that we can reuse the upload params struct in res_texture.cpp as well.